omnibus/files/server-ctl-cookbooks/infra-server/resources/opensearch_index.rb

```diff
@@ -0,0 +1,104 @@
+#
+# Copyright:: Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+provides :opensearch_index
+
+property :index_name, String, name_property: true
+
+property :server_url, String
+
+property :index_definition, Hash
+
+property :opensearch_auth, Hash
+
+property :search_engine_url, String
+
+action :create do
+  #create_opensearch_user
+  unless retry_index_exists?(4)
+    converge_by "Creating opensearch index #{new_resource.index_name}" do
+      search_engine_server.put(new_resource.index_name, new_resource.index_definition, new_resource.opensearch_auth)
+    end
+  end
+end
+
+action :destroy do
+  if index_exists?
+    converge_by "Deleting elasticsearch index #{new_resource.index_name}" do
+      search_engine_server.delete(new_resource.index_name, new_resource.opensearch_auth)
+    end
+  end
+end
+
+action_class do
+  # Tobe removed after review
```

Comment:
This looks like the right thing to do, so I'm not sure why we're not.  Could you provide some context?

Reply:
In the Internal scenario, The user creation should be by updating the security config file not with the API call for fresh install.
In the External scenario, We can't have the full control so expecting the credentials from them and using the same in Erchef application.
Hence I commented out this creating funtion.I will be removing this code block.